### PR TITLE
Model lexer: Fix remaining issues

### LINKED ISSUE #15679: antlr's lack of unicode is disturbing
It's impossible to correctly form XID_start and XID_continue in antlr, so for now the reference lexer just ignores unicode entirely.

### LINKED ISSUE #15878: Weirdness with shebang between model and rustc
Tracking issue for the lexer test failing `test/run-pass/shebang.rs`

### LINKED ISSUE #15882: Model lexer doesn't handle CRLF correctly
Tracking issue for the lexer test failing `test/run-pass/lexer-crlf-line-endings-string-literal-doc-comment.rs`

### LINKED ISSUE #15883: Model lexer is still wrong
Tracking issue for all known problems with the lexer.

Todo:

- [ ] https://github.com/rust-lang/rust/issues/15679 antlr's lack of unicode is disturbing
- [x] https://github.com/rust-lang/rust/issues/15877 Model lexer doesn't handle range correctly
- [ ] https://github.com/rust-lang/rust/issues/15878 Weirdness with shebang between model and rustc
- [x] https://github.com/rust-lang/rust/issues/15879 Model lexer doesn't handle question marks
- [ ] https://github.com/rust-lang/rust/issues/15881 Model lexer doesn't handle ... something?
- [ ] https://github.com/rust-lang/rust/issues/15882 Model lexer doesn't handle CRLF correctly
- [ ] Remove any remaining `ignore-lexer-test`s.

### PR DESCRIPTION
**Pull request by pczarn**
Fixes #15679
Fixes #15878
Fixes #15882
Closes #15883

### CODE REVIEW

**Comment by pczarn** [comment]
cc @fhahn

**Comment by fhahn** [comment]
@ pczarn :+1:, thanks for taking the time to complete this :)

**Comment by alexcrichton** [comment]
cc @cmr 

Looks like there are some tidy errors, but nice work!

**Comment by pczarn** [comment]
Updated

**Comment by cmr** [comment]
Nice work! I'm pleased to see this :)

@bors r+ rollup

File: src/grammar/verify.rs [hunk 1]
```diff
@@ -234,24 +251,39 @@ fn tok_cmp(a: &token::Token, b: &token::Token) -> bool {
...
+    }
+
+    // Rust's lexer
+    let mut code = String::new();
+    File::open(&Path::new(&filename)).unwrap().read_to_string(&mut code).unwrap();
 
-    let mut token_file = File::open(&Path::new(args[2]));
-    let token_map = parse_token_list(token_file.read_to_string().unwrap());
+    let surrogate_pairs_pos: Vec<usize> = code.chars().enumerate()
+                                                     .filter(|&(_, c)| c as usize > 0xFFFF)
```
Comment (tamird) [inline src/grammar/verify.rs]: https://doc.rust-lang.org/nightly/std/iter/trait.Iterator.html#method.filter_map

**Pull request merged**